src/main/java/com/project/stockexchangeappbackend/repository/StockRepository.java

```diff
@@ -2,19 +2,32 @@
 
 import com.project.stockexchangeappbackend.entity.Stock;
 import com.project.stockexchangeappbackend.util.timemeasuring.DBQueryMeasureTime;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.lang.Nullable;
 import org.springframework.stereotype.Repository;
 
 import java.util.Optional;
 
 @Repository
-public interface StockRepository extends JpaRepository<Stock, Long> {
+public interface StockRepository extends JpaRepository<Stock, Long>, JpaSpecificationExecutor<Stock> {
 
+    @Override
     @DBQueryMeasureTime
     <S extends Stock> S save(S s);
 
     @Override
     @DBQueryMeasureTime
     Optional<Stock> findById(Long id);
 
+    @Override
+    @DBQueryMeasureTime
+    Page<Stock> findAll(@Nullable Specification<Stock> var1, Pageable var2);
+
+    @DBQueryMeasureTime
+    Optional<Stock> findByAbbreviation(String abbreviation);
```

Comment:
What about using findByAbbreviationIgnoreCase()? Abbreviation must be unique any way.